### DISCUSSION TITLE
Add humidity sensor/attribute to climate entity

### DISCUSSION THREAD

**Discussion by ZaGaPonG** [jmcollin78/versatile_thermostat#633] [Ideas]
Hello @jmcollin78,

First, thanks for this really good addon 🥇 

Could you envisage to manage to "current_humidity" attribute from climate entity please?

In fact, as my room temp sensors added to each thermostat deliver humidity as well that could be nice if we could link this humidity sensor to the climate entity. I have a climate entity delivered by Tado addon which display the twice: current humidity and current temp.

Exemple with the Tado climate on the top and climate entity created with your addon below:
<img width="336" alt="image" src="https://github.com/user-attachments/assets/ff8834b3-07e1-4138-88d5-f595fce38001">

Thanks in advance

**Comment by jmcollin78** [comment]
Hello @ZaGaPonG ,

It was already claimed here: https://github.com/jmcollin78/versatile_thermostat/issues/75 but refused because of lack of real need.
I suppose you are in `over_climate` Vtherm and if I understand, you just want that Vtherm expose the underlying humidity ?
Am I right ?

**Reply by ZaGaPonG** [reply]
Hello @jmcollin78,

Yes, that's it 👍

In addition, that could open some possibilities because the temp slope is highly link to humidity but that's not my need at the moment.

Thanks

**Reply by jmcollin78** [reply]
I can't integrate humidity because this really not a standard to have this information in a climate entity.

I will transform into an issue to expose the underlying humidity if available.

**Comment by KipK** [comment]
It is a standard for Google Home, and hass climate entity has a current_humidity attribute.
Google Home can gets humidity value from the thermostat. 
If there's a thermostat in the same room as humidity sensor and you ask google home for humidity, then google home only search the humidity value in the thermostat ( I know this is bad implementation probably ) and say it doesn't find it.

That's why I've added this prop to my former mqtt thermostat. Now I miss the feature using versatile. If you can add it that would be great. 👍

**Reply by jmcollin78** [reply]
> It is a standard for Google Home, and hass climate entity has a current_humidity attribute.

I have implemented the current_humidity in a VTherm `over_climate`. So if the underlying climate have the humidity it should be exposed by VTherm. I have nothing to test that, so I'm not sure it is working.

**Reply by KipK** [reply]
unfortunately my vterms are over_switch not over_climate. The only solutions I have for now is doing a template for each vterms to just add the current_humidity attribute.
Having the possiblity to add a humidity sensor to a vtherm would be better.

**Reply by jmcollin78** [reply]
> Having the possiblity to add a humidity sensor to a vtherm would be better.

Certainly how this will be used ? If it is only to expose the humidity you already have it into your sensor. In `over_climate` I understand this can be useful to expose the underlying humidity but for `over_switch` or `over_valve` I don't understand the benefit (and the user have to configure one more parameter).  I really want to avoid adding always more feature that requires configuration attributes and that will generates more questions and issues. Since the humidity is not use for VTherm calculation, I don't intend to add it.

**Reply by KipK** [reply]
As explained above, you can't use a humidity sensor with Google Home if there's already a thermostat defined in the same room. 
If there's only the sensor, then GH can report humidity in the room, but if there's a thermsotat, it ignores the sensor and will search humidity in the thermostat device, failling to find and then report an error. 

I now this is just to solve a bug on GH side, but this bug is present for years now and Google always ignored to fix it. Even their matter implemenation still has the same behavior... 
That's why it was added on Jeedom thermostat and on new HASS matter bridge

Here is how it looks in GH thermostat UI when humidity attribute is present in the matter bridged climate entity.
![Screenshot_20241201-225952](https://github.com/user-attachments/assets/7d27be97-1903-4ed6-9e3a-aae2fa172820)

**Reply by jmcollin78** [reply]
Understood. And I can confirm I will not integrate the humidity in VTherm as a workaround of the Google Home issue.
Sorry for that but the rule exposed above is certainly the most important for the futur (have something sustainable and the simpler as possible for the user).

**Comment by MBRetrofit** [comment]
The temperature refers to the actual measurement of how hot or cold the air is, typically measured in degrees Celsius (°C) or Fahrenheit (°F). This is what you see on a thermometer.

The feels like temperature, often referred to as the "apparent temperature," takes into account additional factors that affect human perception of temperature. This can include:

**Humidity**: High humidity can make it feel warmer than the actual temperature because it reduces the body's ability to cool itself through sweat evaporation. Conversely, low humidity can make it feel cooler.
**Wind**: Wind can increase the cooling effect on the body, making it feel colder than the actual temperature. This is often referred to as the "wind chill" effect.
**Sunlight**: Direct sunlight can make it feel warmer than the actual air temperature, while shade can have the opposite effect.
In summary, while temperature gives you a straightforward reading of the air's warmth or coldness, the "feels like" temperature accounts for environmental conditions that influence how we experience that temperature.

Although the Wind and Sunshine is not common into a interior heated space, the Humidity becomes the most important factor that can influence the Feels Like Temperature.

Humans generally feel comfortable between temperatures of 22 °C to 27 °C and a relative humidity of 40% to 60%.

For example, at 25% humidity with 22 °C, the skin will consume the body heat to evaporate the perspiration with ease and your skin will be cooler compared with room temperature and you will "feel like another temperature". 

A higher percentage of humidity will saturate the perspiration evaporation and again your body will feel the wrong comfort.


There is a lot of ways to calculate the Feel Like Temp, I personally subtracted the Feel Like from Real Temp from Openweather

"{{ '%0.2f' | format(states('sensor.openweathermap_feels_like_temperature') | float - states('sensor.openweathermap_temperature') | float) }}"

And the obtained value I added over the External Temperature Sensor value. So my external temp value is always modified according with this value.

**Comment by dekimsey** [comment]
I think humidity as part of the thermostat makes a lot of sense, both in taking it into account when computing temperature's effect on "how it feels" but also, _simply as it's own feature_. I posted a separate discussion, [[req] Support for defining a humidistat to control a humidifier #750](https://github.com/jmcollin78/versatile_thermostat/discussions/750), on this as I didn't want to derail this one with something tangential and possibly off-topic. Just cross-posting since GH doesn't seem to do that natively in discussions.
